test/test_launcher.rb

```diff
@@ -67,64 +67,60 @@ def test_pid_file
     tmp_file = Tempfile.new("puma-test")
     tmp_path = tmp_file.path
     tmp_file.close!
-
     conf = Puma::Configuration.new do |c|
       c.pidfile tmp_path
     end
-
     launcher(conf).write_state
-
     assert_equal File.read(tmp_path).strip.to_i, Process.pid
-
     File.unlink tmp_path
   end
 
   def test_state_permission_0640
-    tmp_file = Tempfile.new("puma-state")
-    tmp_path = tmp_file.path
-    tmp_permission = 0640
-
-    conf = Puma::Configuration.new do |c|
-      c.state_path tmp_path
-      c.state_permission tmp_permission
-    end
+    Dir.mktmpdir do |dir|
+      tmp_file = Tempfile.new("puma-state")
```

Comment:
I think you need to make sure it's inside the temporary directory.
```suggestion
      tmp_file = Tempfile.new(File.join(dir, "puma-state"))
```

Reply:
thanks updated